erpnext/vehicles/doctype/vehicle_booking_order/vehicle_booking_order.py

```diff
@@ -763,28 +782,50 @@ def get_next_document(vehicle_booking_order, doctype):
 	doc.check_cancelled(throw=True)
 
 	if doctype == "Vehicle Receipt":
-		return get_vehicle_receipt(doc)
+		return make_vehicle_receipt(doc)
 	elif doctype == "Vehicle Receipt Return":
-		return get_vehicle_receipt_return(doc)
+		return make_vehicle_receipt_return(doc)
 	elif doctype == "Vehicle Delivery":
-		return get_vehicle_delivery(doc)
+		return make_vehicle_delivery(doc)
 	elif doctype == "Vehicle Delivery Return":
-		return get_vehicle_delivery_return(doc)
+		return make_vehicle_delivery_return(doc)
 	elif doctype == "Vehicle Invoice":
-		return get_vehicle_invoice(doc)
+		return make_vehicle_invoice(doc)
 	elif doctype == "Vehicle Invoice Delivery":
-		return get_vehicle_invoice_delivery(doc)
+		return make_vehicle_invoice_delivery(doc)
 	elif doctype == "Vehicle Transfer Letter":
-		return get_vehicle_transfer_letter(doc)
+		return make_vehicle_transfer_letter(doc)
 	elif doctype == "Vehicle Registration Order":
-		return get_vehicle_registration_order(doc)
+		return make_vehicle_registration_order(doc)
 	elif doctype == "Project":
-		return get_pdi_repair_order(doc)
+		return make_pdi_repair_order(doc)
 	else:
 		frappe.throw(_("Invalid DocType"))
 
-
-def get_vehicle_receipt(source):
+@frappe.whitelist()
+def make_vehicle_delivery_gate_pass(source_name, target_doc=None):
+	def set_missing_values(source, target):
```

Comment:
add check_if_doc_exists and see how `make_vehicle_delivery` works .. use the same method to create gate pass instead of creating your own method